packages/eas-cli/src/metadata/apple/tasks/app-version.ts

```diff
@@ -156,3 +194,58 @@ async function resolveVersionAsync(
     versionIsFirst: versions.length === 1,
   };
 }
+
+/**
+ * Search for editable app store versions that matches the `versionString` option.
+ * When nothing is found, it will return `null`, and a new version should be created.
+ */
+async function findEditAppStoreVersionAsync(
+  app: App,
+  options: { version: string; platform: Platform }
+): Promise<AppStoreVersion | null> {
+  if (options.version) {
+    const versions = await app.getAppStoreVersionsAsync({
+      query: {
+        limit: 200,
+        filter: {
+          platform: options.platform,
+          appStoreState: [
+            AppStoreState.PREPARE_FOR_SUBMISSION,
+            AppStoreState.DEVELOPER_REJECTED,
+            AppStoreState.REJECTED,
+            AppStoreState.METADATA_REJECTED,
+            AppStoreState.WAITING_FOR_REVIEW,
+            AppStoreState.INVALID_BINARY,
+          ].join(','),
+        },
+      },
+    });
+
+    const version = versions.find(model => model.attributes.versionString === options.version);
+    if (version) {
+      return version;
+    }
```

Comment:
Maybe a `debug` log showing available versions could be useful here.

Reply:
I need to add debug in some other places, but that sounds like a good idea